frontend/src/lib/components/ExportButton/exporterLogic.ts

```diff
@@ -0,0 +1,107 @@
+import { delay } from 'lib/utils'
+import posthog from 'posthog-js'
+import { teamLogic } from 'scenes/teamLogic'
+import { lemonToast } from '../lemonToast'
+
+import type { exporterLogicType } from './exporterLogicType'
+
+const POLL_DELAY_MS = 1000
+const MAX_POLL = 10
+
+export interface ExporterLogicProps {
+    dashboard_id?: number
+    insight_id?: number
+}
+
+export enum ExporterFormat {
+    PNG = 'image/png',
+    CSV = 'text/csv',
+    PDF = 'application/pdf',
+}
+
+export const exporterLogic = kea<exporterLogicType<ExporterFormat, ExporterLogicProps>>([
+    path(['lib', 'components', 'ExportButton', 'ExporterLogic']),
+    props({} as ExporterLogicProps),
+    key(({ dashboard_id, insight_id }) => {
+        return `dash:${dashboard_id}::insight:${insight_id}`
+    }),
+    actions({
+        exportItem: (exportFormat: ExporterFormat, successCallback?: () => void) => ({ exportFormat, successCallback }),
+        exportItemSuccess: true,
+        exportItemFailure: true,
+    }),
+
+    reducers({
+        exportInProgress: [
+            false,
+            {
+                exportItem: () => true,
+                exportItemSuccess: () => false,
+                exportItemFailure: () => false,
+            },
+        ],
+    }),
+
+    listeners(({ actions, props }) => ({
+        exportItem: async ({ exportFormat, successCallback }) => {
+            lemonToast.info(`Export started...`)
+
+            const trackingProperties = {
+                export_format: exportFormat,
+                dashboard: props.dashboard_id,
+                insight: props.insight_id,
+                total_time_ms: 0,
+            }
+            const startTime = performance.now()
+
+            try {
+                let exportedAsset = await api.create(`api/projects/${teamLogic.values.currentTeamId}/exports`, {
+                    export_format: exportFormat,
+                    dashboard: props.dashboard_id,
+                    insight: props.insight_id,
+                })
+
+                if (!exportedAsset.id) {
+                    throw new Error('Missing export_id from response')
+                }
+
+                const downloadUrl = api.exports.determineExportUrl(exportedAsset.id)
+
+                let attempts = 0
+
+                // TODO: Move this to some sort of Kea-style polling?
+                while (attempts < MAX_POLL) {
+                    attempts++
+
+                    if (exportedAsset.has_content) {
+                        actions.exportItemSuccess()
+                        lemonToast.success(`Export complete.`)
+                        successCallback?.()
+
+                        window.open(downloadUrl, '_blank')
+
+                        trackingProperties.total_time_ms = performance.now() - startTime
+                        posthog.capture('export succeeded', trackingProperties)
+
+                        return
+                    }
+
+                    await delay(POLL_DELAY_MS)
```

Comment:
Can do, although given that the API will hold the request open for up to 10 seconds anyways (thats how long it will wait for the celery task to complete), I opted for a second delay as it is probably queued or stuck. 

Either way good points